test/study-webidl.js

```diff
@@ -361,12 +391,318 @@ interface mixin MyNamespaceMixin {};
   attribute boolean hasTVToo;
 };
 `);
-    assert.deepEqual(report[0]?.name, 'unknownExtAttr');
-    assert.deepEqual(report[0].message, `Unknown extended attribute "UnknownExtAttr" used in definition of "MyRoom"`);
-    assert.deepEqual(report[1]?.name, 'unknownExtAttr');
-    assert.deepEqual(report[1].message, `Unknown extended attribute "SuperUnknownExtAttr" used in definition of "MyLivingRoom"`);
-    assert.deepEqual(report[2]?.name, 'unknownExtAttr');
-    assert.deepEqual(report[2].message, `Unknown extended attribute "SuperUnknownExtAttr" used in definition of "MyBedRoom"`);
-    assert.deepEqual(report.length, 3);
+    assertEqual(report.length, 3);
+    assertEqual(report, 0, 'name', 'unknownExtAttr');
+    assertEqual(report, 0, 'message', `Unknown extended attribute "UnknownExtAttr" used in definition of "MyRoom"`);
+    assertEqual(report, 1, 'name', 'unknownExtAttr');
+    assertEqual(report, 1, 'message', `Unknown extended attribute "SuperUnknownExtAttr" used in definition of "MyLivingRoom"`);
+    assertEqual(report, 2, 'name', 'unknownExtAttr');
+    assertEqual(report, 2, 'message', `Unknown extended attribute "SuperUnknownExtAttr" used in definition of "MyBedRoom"`);
+  });
+
+
+  it('reports overloads across definitions (partial, same spec)', async () => {
+    const report = await analyzeIdl(`
+[Global=Home,Exposed=*]
+interface MyHome {
+  undefined overload();
+  undefined overload(DOMString thing);
+};
+
+partial interface MyHome {
+  undefined overload(DOMString thing, boolean asap);
+};
+
+[Exposed=*]
+interface MyPartialHome {};
+
+partial interface MyPartialHome {
+  Promise<DOMString> overload();
+};
+
+partial interface MyPartialHome {
+  Promise<DOMString> overload(DOMString thing);
+};
+`);
+    assertEqual(report.length, 2);
+    assertEqual(report, 0, 'name', 'overloaded');
+    assertEqual(report, 0, 'message', '"operation overload" in partial interface "MyHome" overloads an operation defined in interface "MyHome"');
+    assertEqual(report, 1, 'name', 'overloaded');
+    assertEqual(report, 1, 'message', '"operation overload" in partial interface "MyPartialHome" overloads an operation defined in another partial interface "MyPartialHome"');
+  });
+
+
+  it('reports overloads across definitions (partial, different specs)', async () => {
+     const crawlResult = toCrawlResult(`
+[Global=Home,Exposed=*]
+interface MyHome {
+  undefined overload();
+  undefined overload(DOMString thing);
+};
+
+[Exposed=*]
+interface MyPartialHome {};
+
+partial interface MyPartialHome {
+  Promise<DOMString> overload();
+};
+`).concat(toCrawlResult(`
```

Comment:
the `concat` pattern is used often enough that it may be better to redefine `toCrawlResult` to take two IDL declaration and assign the spec URL accordingly.